### PR TITLE
check for empty arrays and return zero confidence

### PR DESCRIPTION
**Pull request by mnutt**
Ensures that passing `[]` to the initializer doesn't blow up, and instead returns zero confidence.

### CODE REVIEW

File: spec/u_mann_whitney_spec.rb [hunk 1]
```diff
@@ -6,29 +6,43 @@
...
+      it 'has correct value for z and exact probability' do
+        expect(@u.probability_z).to be_within(0.001).of(0.027)
+        expect(@u.probability_exact).to be_within(0.001).of(0.028)
+      end
     end
 
-    it 'has correct value for z and exact probability' do
-      expect(@u.probability_z).to be_within(0.001).of(0.027)
-      expect(@u.probability_exact).to be_within(0.001).of(0.028)
+    describe "with invalid values" do
```
Comment (shyshy) [inline spec/u_mann_whitney_spec.rb:36]: Not familiar with the two libraries in question (`Daru` and `UMannWhiteny`), but my question is, is passing an empty array actually an "invalid" value? i.e. is passing an empty array into `UMannWhiteny.new` supposed to error because you would _never_ pass an empty array when interacting with the library?

Or should `Daru` never be supplied with an empty dataset in its initializer _or_ should it actually return an empty array instead of `nil`?

In other words are we correcting a behavior / fixing a bug or are we adjusting it to guard against an edgecase that we have in our system? The description of this spec makes me think it's the former, but, I wanna make sure for my own understanding.

**Review by shyshy — APPROVED** [review]
LGTM just had one question about the description for a spec.

File: spec/u_mann_whitney_spec.rb (hunk 1 above)
Comment (mnutt) [reply spec/u_mann_whitney_spec.rb:36]: I don't know enough about the underlying algorithm to be totally sure, but we used to pass empty arrays to it and it at least didn't blow up.
